pkg/framework/machines.go

```diff
@@ -44,6 +44,28 @@ func GetMachine(c client.Client, name string) (*mapiv1beta1.Machine, error) {
 	return machine, nil
 }
 
+// MachinesPresent looks for matching machine in a predefined list
+func MachinesPresent(existingMachines []*mapiv1beta1.Machine, machines ...*mapiv1beta1.Machine) bool {
+	if len(existingMachines) < len(machines) {
+		return false
+	}
+	for _, machine := range machines {
```

Comment:
this nested loops would be cheaper in terms of complexity if you have two independent loops over a map and hash objects by UID.